nucliadb_node/src/shards/shard_writer.rs

```diff
@@ -436,6 +436,13 @@ impl ShardWriter {
         }
     }
 
+    #[tracing::instrument(skip_all)]
+    pub fn merge(&self) -> NodeResult<MergeMetrics> {
+        let _lock = self.gc_lock.blocking_lock();
```

Comment:
Why do you want this? This will prevent GC and replication while force_merging. GC is already prevented by the slock/elock, so taking this lock here basically just blocks replication while merging. Any particular reason why?